gui/web/src/components/atoms/Button/Button.js

```diff
@@ -34,8 +45,34 @@ class Button extends Component {
     variant: PropTypes.string,
     onClick: PropTypes.func,
     loading: PropTypes.bool,
-    disabled: PropTypes.bool
+    disabled: PropTypes.bool,
+    trackEvent: PropTypes.func,
+    eventName: PropTypes.string,
   };
+  
+  sendMetricEvent() {
+    if (this._asyncRequests["sendMetricEvent"]) {
+      return
+    }
+    
+    var _this = this
+    // TODO DS populate the eventData.
+    this._asyncRequests["sendMetricEvent"] = sendMetricEvent(this.props.baseUrl, this.props.eventName, {}).then(resp => {
+      if (!_this._asyncRequests["sendMetricEvent"]) {
+        // if it has been deleted it means we don't want to process the result
+        return
+      }
+      
+      delete _this._asyncRequests["sendMetricEvent"];
+      
+      // TODO DS Determine how to process resp
```

Comment:
if resp is success then do nothing, if there's an error then just console.log it for now.